### PR TITLE
OCPBUGS-86886: modernize dependency of k8s imports

### PR DESCRIPTION
**Pull request by rikatz**
This PR removes dependency of imports that were once necessary but were moved to a proper package, reducing the surface of import of libraries on route-controller-manager

## Summary by CodeRabbit

* **New Features**
  * Added an in-memory CIDR-backed IP range allocator with allocate/release/next/has operations.
* **Documentation**
  * Added README documenting the IP allocator behavior and constraints.
* **Tests**
  * Added unit tests covering allocator behavior, errors, and edge cases.
* **Chores**
  * Removed several unused dependencies.
* **Refactor**
  * Updated internal Kubernetes client/controller integration to modern scheme/queue handling.

### CODE REVIEW

**Comment by bentito** [comment]
I think this PR takes care of everything in `pkg/route/ingress/` but we still leave a dependency in `pkg/route/ingressip/`
We could ignore that as `k8s.io/kubernetes/pkg/registry/core/service/ipallocator` has not been lifted out into place truly meant for external use. But then we keep the dependency on k8s.io/kubernetes from a scanner perspective.

File: pkg/utils/ipallocator/ipallocator.go [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+		return false, 0
+	}
+	return true, offset
+}
+
+// calculateIPOffset calculates the integer offset of ip from base such that
+// base + offset = ip. It requires ip >= base.
+func calculateIPOffset(base *big.Int, ip net.IP) int {
+	return int(big.NewInt(0).Sub(netutils.BigForIP(ip), base).Int64())
+}
```
[the remark this answers is not in the document]
Comment (rikatz) [reply pkg/utils/ipallocator/ipallocator.go:184]: @coderabbitai this code was copied from the dropped vendor vendor/k8s.io/kubernetes/pkg/registry/core/service/ipallocator and I don't wanna change any behavior

File: pkg/route/ingressip/service_ingressip_controller.go [hunk 2]
```diff
@@ -18,15 +18,14 @@ import (
...
 	kv1core "k8s.io/client-go/kubernetes/typed/core/v1"
 	"k8s.io/client-go/tools/cache"
 	"k8s.io/client-go/tools/record"
 	"k8s.io/client-go/util/workqueue"
-	"k8s.io/kubernetes/pkg/api/legacyscheme"
-	"k8s.io/kubernetes/pkg/controller"
-	"k8s.io/kubernetes/pkg/registry/core/service/allocator"
-	"k8s.io/kubernetes/pkg/registry/core/service/ipallocator"
+
+	"github.com/openshift/route-controller-manager/pkg/utils/ipallocator"
```
Comment (rikatz) [inline pkg/route/ingressip/service_ingressip_controller.go:28]: this could be github.com/cilium/ipam which is also a copy, we can consider it

**Comment by rikatz** [comment]
/retest-required
@coderabbitai I want you to verify the dropped dependency from ipallocator and the fact we copied the code, and verify if the code semantics that we are using now is equal, works fine, is not a breaking change, do a line by line analysis

**Comment by bentito** [comment]
/lgtm
latest commits make this all set to go

**Comment by melvinjoseph86** [comment]
Verified on cluster-bot cluster

```
➜  oc get pods --all-namespaces | grep route-controller-manager
openshift-route-controller-manager                 route-controller-manager-b975c6944-m5tx8                                    1/1     Running     0              50m
openshift-route-controller-manager                 route-controller-manager-b975c6944-mtq4v                                    1/1     Running     0              50m
openshift-route-controller-manager                 route-controller-manager-b975c6944-xzvlq                                    1/1     Running     0              50m
➜  oc get clusterversion
NAME      VERSION                                               AVAILABLE   PROGRESSING   SINCE   STATUS
version   5.0.0-0-2026-06-11-163220-test-ci-ln-yz3r8zb-latest   True        False         45m     Cluster version is 5.0.0-0-2026-06-11-163220-test-ci-ln-yz3r8zb-latest
```

Ran regression on ingress and route case and no issues found
/verified by @mjoseph
